### PR TITLE
Create `CalculateRestaurantTotalRevenue(restaurantId)` Database Function and method

### PR DESCRIPTION
**Pull request by Reema-Khaseeb**

### CODE REVIEW

File: RestaurantReservation.Db/Configurations/RestaurantConfiguration.cs [hunk 1]
```diff
@@ -10,6 +10,9 @@ public void Configure(EntityTypeBuilder<Restaurant> builder)
         {
             // Primary Key
             builder.HasKey(r => r.RestaurantId);
+            builder
```
Comment (sabahb3) [inline RestaurantReservation.Db/Configurations/RestaurantConfiguration.cs:13]: Should we add it to the model?

File: RestaurantReservation.Db/Repositories/RestaurantRepository.cs [hunk 2]
```diff
@@ -54,5 +54,15 @@ public async Task DeleteRestaurantAsync(int restaurantId)
                 await _context.SaveChangesAsync();
             }
         }
+
+        public async Task<decimal> CalculateRestaurantTotalRevenueAsync(int restaurantId)
+        {
+            var result = await _context.Restaurants
+                .FromSqlInterpolated($"SELECT dbo.CalculateRestaurantTotalRevenue({restaurantId}) AS TotalRevenue")
```
Comment (sabahb3) [inline RestaurantReservation.Db/Repositories/RestaurantRepository.cs:61]: Should we Use EF core and migration?
I mean the requirement is to be familiar with them not to have raw sql

Comment (Reema-Khaseeb) [reply RestaurantReservation.Db/Repositories/RestaurantRepository.cs:61]: Yes, It's a requirement:

> Create a database function to **`Calculate the total revenue generated by a specific restaurant`** using Entity Framework Core migrations.
> Implement the function call in your **`RestaurantReservation.Db`** project, making it accessible from your application.

File: RestaurantReservation.Db/Configurations/RestaurantConfiguration.cs (hunk 1 above)
Comment (Reema-Khaseeb) [reply RestaurantReservation.Db/Configurations/RestaurantConfiguration.cs:13]: It didn't work until I did this

**Pull request merged**